zipline/algorithm.py

```diff
@@ -636,16 +649,31 @@ def run(self, data=None, overwrite_sim_params=True):
                         copy_panel.items
                     )
                 )
-                equity_daily_reader = PanelDailyBarReader(
-                    self.trading_schedule.all_execution_days,
-                    copy_panel,
-                )
-                self.data_portal = DataPortal(
-                    self.asset_finder,
-                    self.trading_schedule,
-                    first_trading_day=equity_daily_reader.first_trading_day,
-                    equity_daily_reader=equity_daily_reader,
-                )
+
+                if self.sim_params.data_frequency == 'daily':
```

Comment:
Actually, the two cases call for two different keyword arguments: `equity_minute_reader` and `equity_daily_reader`. Any way to resolve that?